### PR TITLE
add stats csv converter to postprocess

### PR DESCRIPTION
**Pull request by emlynjdavies**

### CODE REVIEW

File: pysilcam/postprocess.py [hunk 1]
```diff
@@ -900,6 +900,18 @@ def stats_to_xls_png(config_file, stats_filename, oilgas=outputPartType.all):
...
+def statscsv_to_statshdf(stats_file):
+    '''Convert old STATS.csv file to a STATS.h5 file
+
+    Args:
+        stats_file              : filename of stats file
+    '''
+    stats = pd.read_csv(stats_file)
+    new_stats_file = stats_file.replace('-STATS.csv', '-STATS.h5')
+    stats.to_hdf(new_stats_file, 'ParticleStats/stats', mode='r+', format='t',
+                      data_columns=True)
```
Comment (tankefugl) [inline pysilcam/postprocess.py:912]: E127: continuation line over-indented for visual indent

**Review by tankefugl — CHANGES_REQUESTED** [review]
[Lintly](https://github.com/grantmcconnaughey/Lintly) has detected code quality issues in this pull request.

**Review by tankefugl — CHANGES_REQUESTED** [review]
[Lintly](https://github.com/grantmcconnaughey/Lintly) has detected code quality issues in this pull request.

File: pysilcam/tests/test_stats_convert.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    with tempfile.TemporaryDirectory() as tempdir:
+        print('tempdir:', tempdir, 'created')
+        local_filename = 'STN07-STATS.csv'
+        stats.to_csv(local_filename)
+        statscsv_to_statshdf(local_filename)
+        assert os.path.isfile(local_filename), ('STATS.h5 file not created.')
+
+        stats = pd.read_hdf(local_filename, 'ParticleStats/stats')
+        new_stats_shape = np.shape(stats)
+        assert (original_stats_shape == new_stats_shape), ('mismatch in data between csv and h5 stats')
```
Comment (tankefugl) [inline pysilcam/tests/test_stats_convert.py:21]: W292: no newline at end of file

File: pysilcam/tests/test_stats_convert.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+from pysilcam.postprocess import statscsv_to_statshdf
+import pandas as pd
+import tempfile
+import os
+import numpy as np
+
+def test_stats_convert():
```
Comment (tankefugl) [inline pysilcam/tests/test_stats_convert.py:7]: E302: expected 2 blank lines, found 1

File: pysilcam/postprocess.py [hunk 4]
```diff
@@ -70,30 +74,31 @@ def get_size_bins():
...
     # pre-allocate
-    bin_mids_um = np.zeros((52),dtype=np.float64)
+    bin_mids_um = np.zeros((52), dtype=np.float64)
 
     # define the middle of the smallest bin (same as LISST-100x type-c)
     bin_mids_um[0] = 2.72
 
     # loop through 53 size classes and calculate the bin mid-points
-    for I in np.arange(1,52,1):
+    for I in np.arange(1, 52, 1):
```
Comment (tankefugl) [inline pysilcam/postprocess.py:94]: E741: ambiguous variable name 'I'

File: pysilcam/postprocess.py [hunk 5]
```diff
@@ -70,30 +74,31 @@ def get_size_bins():
...
     # pre-allocate
-    bin_limits_um = np.zeros((53),dtype=np.float64)
+    bin_limits_um = np.zeros((53), dtype=np.float64)
 
     # define the upper limit of the smallest bin (same as LISST-100x type-c)
     bin_limits_um[0] = 2.72 * 0.91
 
     # loop through 53 size classes and calculate the bin limits
-    for I in np.arange(1,53,1):
+    for I in np.arange(1, 53, 1):
```
Comment (tankefugl) [inline pysilcam/postprocess.py:83]: E741: ambiguous variable name 'I'

**Review by tankefugl — CHANGES_REQUESTED** [review]
[Lintly](https://github.com/grantmcconnaughey/Lintly) has detected code quality issues in this pull request.

File: pysilcam/tests/test_stats_convert.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        statscsv_to_statshdf(local_csv_filename)
+        local_hdf_filename = local_csv_filename.replace('-STATS.csv', '-STATS.h5')
+        print(local_hdf_filename)
+        assert os.path.isfile(local_hdf_filename), ('STATS.h5 file not created.')
+
+        stats = pd.read_hdf(local_hdf_filename, 'ParticleStats/stats')
+        new_stats_shape = np.shape(stats)
+        print('original_stats_shape', original_stats_shape)
+        print('new_stats_shape', new_stats_shape)
+        assert (original_stats_shape == new_stats_shape), ('mismatch in data between csv and h5 stats')
```
Comment (tankefugl) [inline pysilcam/tests/test_stats_convert.py:25]: W292: no newline at end of file

File: pysilcam/tests/test_stats_convert.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
+from pysilcam.postprocess import statscsv_to_statshdf
+import pandas as pd
+import tempfile
+import os
+import numpy as np
+
+def test_stats_convert():
```
Comment (tankefugl) [inline pysilcam/tests/test_stats_convert.py:7]: E302: expected 2 blank lines, found 1

File: pysilcam/postprocess.py (hunk 4 above)
Comment (tankefugl) [inline pysilcam/postprocess.py:94]: E741: ambiguous variable name 'I'

File: pysilcam/postprocess.py (hunk 5 above)
Comment (tankefugl) [inline pysilcam/postprocess.py:83]: E741: ambiguous variable name 'I'

**Review by tankefugl — CHANGES_REQUESTED** [review]
[Lintly](https://github.com/grantmcconnaughey/Lintly) has detected code quality issues in this pull request.

File: pysilcam/tests/test_stats_convert.py (hunk 6 above)
Comment (tankefugl) [inline pysilcam/tests/test_stats_convert.py:25]: W292: no newline at end of file

File: pysilcam/tests/test_stats_convert.py (hunk 7 above)
Comment (tankefugl) [inline pysilcam/tests/test_stats_convert.py:7]: E302: expected 2 blank lines, found 1

File: pysilcam/postprocess.py (hunk 4 above)
Comment (tankefugl) [inline pysilcam/postprocess.py:94]: E741: ambiguous variable name 'I'

File: pysilcam/postprocess.py (hunk 5 above)
Comment (tankefugl) [inline pysilcam/postprocess.py:83]: E741: ambiguous variable name 'I'

**Review by tankefugl — CHANGES_REQUESTED** [review]
[Lintly](https://github.com/grantmcconnaughey/Lintly) has detected code quality issues in this pull request.

File: pysilcam/postprocess.py (hunk 4 above)
Comment (tankefugl) [inline pysilcam/postprocess.py:94]: E741: ambiguous variable name 'I'

File: pysilcam/postprocess.py (hunk 5 above)
Comment (tankefugl) [inline pysilcam/postprocess.py:83]: E741: ambiguous variable name 'I'

**Review by tankefugl — CHANGES_REQUESTED** [review]
[Lintly](https://github.com/grantmcconnaughey/Lintly) has detected code quality issues in this pull request.

File: pysilcam/postprocess.py [hunk 8]
```diff
@@ -90,9 +90,9 @@ def get_size_bins():
     bin_mids_um[0] = 2.72
 
     # loop through 53 size classes and calculate the bin mid-points
-    for i in np.arange(1, 52, 1):
+    for bin_number in np.arange(1, 52, 1):
         # each bin is 1.18 * larger than the previous
-        bin_mids_um[i] = bin_mids_um[i - 1] * 1.180
+        bin_mids_um[bin_number] = bin_mids_um[bin_number - 1] * 1.180
```
Comment (nepstad) [inline pysilcam/postprocess.py:95]: Perhaps instead: bin_mids_um[1:] = bin_mids_um[:-1] * 1.180

File: pysilcam/postprocess.py [hunk 9]
```diff
@@ -928,12 +929,23 @@ def stats_to_xls_png(config_file, stats_filename, oilgas=outputPartType.all):
...
+def statscsv_to_statshdf(stats_file):
+    '''Convert old STATS.csv file to a STATS.h5 file
+
+    Args:
+        stats_file              : filename of stats file
+    '''
+    stats = pd.read_csv(stats_file, index_col=False)
+    new_stats_file = stats_file.replace('-STATS.csv', '-STATS.h5')
+    with pd.HDFStore(new_stats_file, 'a') as fh:
+        stats.to_hdf(fh, 'ParticleStats/stats', mode='a', format='t', data_columns=True)
```
Comment (nepstad) [inline pysilcam/postprocess.py:946]: mode='a' in to_hdf call is not used when HDFStore is used.

**Review by nepstad — CHANGES_REQUESTED** [review]

File: pysilcam/postprocess.py (hunk 8 above)
Comment (emlynjdavies) [reply pysilcam/postprocess.py:95]: not sure i understand. bin_number is an iterator in the for loop.

**Review by nepstad — APPROVED** [review]

**Review by WillRNaylor — APPROVED** [review]

**Pull request merged**